### DISCUSSION TITLE
[BUG]:[API V3]  Update shop shipping profile returns new error about origin_postal_code

### DISCUSSION THREAD

**Discussion by TungNguyen205** [etsy/open-api#749] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Hi ,
I use api POST shop shipping profile to create a new one with the shipping origin is VIETNAM, the payload:
`{
    "title": "new shipping",
    "min_processing_time": 1,
    "max_processing_time": 2,
    "origin_country_iso": "VN",
    "destination_country_iso": "VN",
    "primary_cost": 1000,
    "secondary_cost": 888,
    "min_delivery_days": 1,
    "max_delivery_days": 2,
    "processing_time_unit": "business_days"
}`
then I get the success message:
`{
    "shipping_profile_id": 171330656318,
    "title": "new shipping",
    "user_id": 605453254,
    "min_processing_days": 1,
    "max_processing_days": 2,
    "processing_days_display_label": "1-2 business days",
    "origin_country_iso": "VN",
    "origin_postal_code": "",
    "profile_type": "manual",
    "is_deleted": false,
    "domestic_handling_fee": 0,
    "international_handling_fee": 0,
    "shipping_profile_destinations": [
        {
            "shipping_profile_destination_id": 31928109603,
            "shipping_profile_id": 171330656318,
            "origin_country_iso": "VN",
            "destination_country_iso": "VN",
            "destination_region": "none",
            "primary_cost": {
                "amount": 100000,
                "divisor": 100,
                "currency_code": "USD"
            },
            "secondary_cost": {
                "amount": 80000,
                "divisor": 100,
                "currency_code": "USD"
            },
            "shipping_carrier_id": 0,
            "mail_class": null,
            "min_delivery_days": 1,
            "max_delivery_days": 2
        }
    ],
    "shipping_profile_upgrades": []
}`

BUT when I edit this shipping profile without or with **origin_postal_code** field, I get the error:
- without origin_postal_code:
`{
    "error": "ZIP Code  is invalid"
}`
- With VIETNAM postal code 700000
`{
    "error": "We currently do not support submitting postal codes for this country: VN"
}`

### Steps to reproduce

1. use curl
`curl --location --request POST 'https://openapi.etsy.com/v3/application/shops/34481652/shipping-profiles' \
--header 'x-api-key: qilngv1u7vjydeq4p2aokq42' \
--header 'Authorization: Bearer 605453254.uWHafzJGni7LbL_VbPREHrDO7BtV3QbODFTDseCAHGp1QOdzklNfgEjsRbDoZ0OfBno-9ntx4zmksbCuJeOyt0n85C' \
--header 'Content-Type: application/json' \
--header 'Cookie: user_prefs=dKtcG6OQNjcPnP_q3R7C23PLy-pjZACCxKu8yjA6Oq80J0eHPCKWAQA.' \
--data-raw '{
    "title": "new shipping",
    "min_processing_time": 1,
    "max_processing_time": 2,
    "origin_country_iso": "VN",
    "destination_country_iso": "VN",
    "primary_cost": 1000,
    "secondary_cost": 800,
    "min_delivery_days": 1,
    "max_delivery_days": 2,
    "processing_time_unit": "business_days"
}'`
Then edit with the specific **id** the system return:
`curl --location --request PUT 'https://openapi.etsy.com/v3/application/shops/34481652/shipping-profiles/171330656318' \
--header 'x-api-key: qilngv1u7vjydeq4p2aokq42' \
--header 'Authorization: Bearer 605453254.OAzaT3F3wb-o3MM2ofLgsBWWBXrBl3Cyjy20IzIlUHH5VwqBXWGIlyoVgzMG1t5mcgFQB5rC5qKPgRFeWrneuwK1gl' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--header 'Cookie: user_prefs=dKtcG6OQNjcPnP_q3R7C23PLy-pjZACCxKu8yjA6Oq80J0eHPCKWAQA.' \
--data-urlencode 'origin_country_iso=VN' \
--data-urlencode 'min_processing_time=1' \
--data-urlencode 'max_processing_time=2' \
--data-urlencode 'processing_time_unit=business_days' \
--data-urlencode 'origin_postal_code=700000'`

### Expected behavior

Hope I can update shipping profile with the specific id

**Comment by lehoangnnx** [comment]
I also have the same problem

**Comment by etsyachristensen** [comment]
We are aware of this issue. We are working on a fix that should be in place early next week.

**Comment by etsyachristensen** [comment]
This should be fixed now. Please let us know if you continue to get errors.

**Comment by amitkumar-248** [comment]
> This should be fixed now. Please let us know if you continue to get errors.

@etsyachristensen still getting same error for previous client. https://github.com/etsy/open-api/issues/433

**Comment by etsyachristensen** [comment]
@amitshuklaCedcoss Can you give me the shop id and the error message you are getting? An updated UUID would be helpful too. Thanks!

**Comment by amitkumar-248** [comment]
> @amitshuklaCedcoss Can you give me the shop id and the error message you are getting? An updated UUID would be helpful too. Thanks!

[shop_id] => 20232052
[shipping_profile_id] => 72471460592
 [error] => There was a problem with /item/weight : Invalid Shipping Data
  [X-Etsy-Request-Uuid] => Array
                        (
                            [0] => Eur1aDVaXZ05UqmqiH58OvHCCfba
                        )
                        
hope this info will help you in debug. 
thanks

**Comment by amitkumar-248** [comment]
@etsyachristensen any luck?

**Comment by amitkumar-248** [comment]
> [error] => There was a problem with /item/weight : Invalid Shipping Data

@etsyachristensen @jasonpincin can you plz re-open this case until it's resolved?

**Comment by etsyachristensen** [comment]
@amitshuklaCedcoss Can you give me the full payload and headers you are sending to the api for this (remove your api key and token)?

**Comment by amitkumar-248** [comment]
```
 [request_headers] => Array
        (
            [Content-Type] => application/x-www-form-urlencoded
            [x-api-key] => API_KEY
            [Authorization] => Bearer ACCESS_TOKEN
        )


 [request_payload] => Array
        (
            [item_weight] => 0.3
            [item_weight_unit] => lb
            [item_length] => 9
            [item_width] => 12
            [item_height] => 1
            [item_dimensions_unit] => in
            [title] => Vivi's Vintage T-Shirt, Maroon
            [description] => Get your own Vivi's Vintage Market shirt! Unisex sizes available in this roomy shirt.             
            **[shipping_profile_id] => 72471460592**          
            [processing_min] => 1
            [processing_max] => 3
            [taxonomy_id] => 482
            [who_made] => i_did
            [when_made] => 2020_2022
            [is_supply] => 0
            [state] => active
        )
)

[return_response] => [       
    **"error" => "There was a problem with /item/weight : Invalid Shipping Data",**    
    "http_code"=> 400,
]

[response_headers] => {
        "Server":
        [
            "Apache"
        ],
        "X-CLOUD-TRACE-CONTEXT":
        [
            "a2bd106185d55824e5fbecdfbb6fdae3/1728672233195263964;o=1"
        ],
        **"X-Etsy-Request-Uuid":
        [
            "EuN2n7iKXXsfXVyYha0gJogc7U3b"
        ],**               
        "Content-Length":
        [
            "76"
        ],
        "Content-Type":
        [
            "application/json"
        ],
        "Date":
        [
            "Fri, 06 May 2022 05:55:57 GMT"
        ],        
    }
```

@etsyachristensen hopes this data will be sufficient for debugging. 
Waiting for your next response. 
Thanks

**Comment by etsyachristensen** [comment]
@amitshuklaCedcoss Is this an `updateListing` call or a `createDraftListing` call?

In update, there is no `processing_min` or `processing_max`. But in create, there's no `state` option. It only creates a draft. You then have to do update to set it to active. Even then, you couldn't take it live without setting price and quantity using the `updateListingInventory` endpoint or by passing those variables to the `createDraftListing`.

I've been trying to recreate your issue and so far haven't been successful. Do I have your permission to log into the shop listed above (20232052) and attempt to make test create/update calls there to try to debug this?

**Comment by amitkumar-248** [comment]
> I've been trying to recreate your issue and so far haven't been successful. Do I have your permission to log into the shop listed above (20232052) and attempt to make test create/update calls there to try to debug this?

@etsyachristensen please try to update the listing (**listing_id: 793723890**), I had tried after removing processing_min and processing_max from updateListing call. 
if you wish you can create test listings, and we will reimburse the listing charge to the client. 
thanks

**Comment by amitkumar-248** [comment]
@etsyachristensen did u tried to regenerate above bug?

**Comment by etsyachristensen** [comment]
@amitshuklaCedcoss Not yet. I've been backlogged on some stuff. Will try to get it done as soon as I can. Thanks!

**Comment by etsyachristensen** [comment]
Oh my goodness. I can't believe I didn't see this before... The problem is that the item_length should be the greatest value of the item dimensions. So in your update you have item_length as 9 and item_width as 12. Swap those values and it will work. So sorry I didn't catch that before. I had to duplicate the problem to see it. I'll look at documentation and errors to see if we can get it to explain this better.

**Comment by amitkumar-248** [comment]
Thank you so much @etsyachristensen 
Gratitude

**Comment by amitkumar-248** [comment]
@etsyachristensen again getting this error on different stores
using [createDraftListing](https://developers.etsy.com/documentation/reference/#operation/createDraftListing) call
[shop_id] => 27068452

error details : 
Client error: `POST https://openapi.etsy.com/v3/application/shops/27068452/listings?language=en` resulted in a `400 Bad Request` response:
[{"path":"\/item\/weight","type":"shipping_profile_no_domestic_option","message":"Invalid Shipping Data"},{"path":"\/ite (truncated...)

```
[error] => Array
                (
                    [0] => Array
                        (
                            [path] => /item/weight
                            [type] => shipping_profile_no_domestic_option
                            [message] => Invalid Shipping Data
                        )

                    [1] => Array
                        (
                            [path] => /item/length
                            [type] => shipping_profile_no_domestic_option
                            [message] => Invalid Shipping Data
                        )

                )


[http_code] => 400
            [response_headers] => Array
                (
                    [Server] => Array
                        (
                            [0] => Apache
                        )

                    [X-CLOUD-TRACE-CONTEXT] => Array
                        (
                            [0] => 094895cf9d3eb5f555b21308588cb889/749116450670418894;o=0
                        )

                    [X-Etsy-Request-Uuid] => Array
                        (
                            [0] => Euho8LLg6YRr60uOfRLz-TRyega2
                        )
)
```

```
[request_payload] = Array
(
    [item_weight] => 18
    [item_weight_unit] => lb
    [item_length] => 59
    [item_width] => 39
    [item_height] => 1
    [item_dimensions_unit] => in
    [quantity] => 1
    [title] => JAYNERIO - Beni Ourain Moroccan Rug
    [shipping_profile_id] => 173076283389
    [taxonomy_id] => 1003
    [processing_min] => 3
    [processing_max] => 5
)

```

I had double-checked that item_length must be greater than width and height.

I hope the above data will help in debugging this issue. 
Thanks

**Comment by etsyachristensen** [comment]
The error you are getting is that you don't have a domestic shipping profile. A domestic option is required when the profile is not set to ship for free domestically. I would check the shipping profile settings.

**Comment by amitkumar-248** [comment]
> using [createDraftListing](https://developers.etsy.com/documentation/reference/#operation/createDraftListing) call [shop_id] => 27068452
> 
> error details : Client error: `POST https://openapi.etsy.com/v3/application/shops/27068452/listings?language=en` resulted in a `400 Bad Request` response: [{"path":"/item/weight","type":"shipping_profile_no_domestic_option","message":"Invalid Shipping Data"},{"path":"/ite (truncated...)

@etsyachristensen client had updated shipping profile for domestic option 
![image](https://user-images.githubusercontent.com/25526103/172654391-065d4233-037a-46c5-8926-cdbd87466a98.png)

Here is the response header for issue debugging

`[shop_id] => 27068452`

```
[response_headers] => Array
                (                 
                    [X-CLOUD-TRACE-CONTEXT] => Array
                        (
                            [0] => 836bd2710af711958f4f8b75d020a711/12990886611862872271;o=0
                        )

                    [X-Etsy-Request-Uuid] => Array
                        (
                            [0] => EuCpk0H5PDYp6KfKDet7AjE8VDd0
                        )
}
```

**Comment by amitkumar-248** [comment]
Hi @etsyachristensen ,
any luck ?

**Comment by etsyachristensen** [comment]
Sorry for the delay. I just tried to look up that ID (27068452) and that shop has no apps attached. I'm not sure how you even got that error message given the shop owner hasn't authorized any apps.

**Comment by amitkumar-248** [comment]
> I'm not sure how you even got that error message given the shop owner hasn't authorized any apps.

may be client had removed my app from shop, but I'm still able to call [getShop](https://developers.etsy.com/documentation/reference/#operation/getShop) , [getShopProductionPartners](https://developers.etsy.com/documentation/reference/#operation/getShopProductionPartners), [getListing](https://developers.etsy.com/documentation/reference/#operation/getListing) etc. 

please try once again to check the original error of **Invalid Shipping Data**

Thanks @etsyachristensen

**Comment by Deepshikha-cedcoss** [comment]
> The problem is that the item_length should be the greatest value of the item dimensions. So in your update, you have item_length as 9 and item_width as 12. Swap those values and it will work

@etsyachristensen getting this error again while the length of the item is greater than other dimensions(height, width). Please help to find out the exact issue for this.

**error_details:**
["last_response"]=>
    string(275) "Client error: `POST https://openapi.etsy.com/v3/application/shops/17815748/listings?language=en` resulted in a `400 Bad Request` response:
  ```
 [message] => Array
                (
                    [0] => Array
                        (
                            [path] => /item/weight
                            [type] => shipping_profile_no_domestic_option
                            [message] => Invalid Shipping Data
                        )

                    [1] => Array
                        (
                            [path] => /item/length
                            [type] => shipping_profile_no_domestic_option
                            [message] => Invalid Shipping Data
                        )

                )
```
```
[request_payload] = Array
(
    [item_weight] => 2000
    [item_weight_unit] => lb
    [item_length] => 50
    [item_width] => 48
    [item_height] => 48
    [item_dimensions_unit] => in
    [quantity] => 999
    [title] => Wood Sphere
    [description] => Created with wood pieces and Ecopoxy™, this custom wood sphere shows the beauty in hand-crafted labor. It is an example of quality and looks great anywhere from a rustic setting to a modern environment.
Images are client projects, we can customize the sphere to your personal liking.

Available in 36 in diameter, starting at $20,000; 42 in diameter starting at $25,000 and 48 in diameter starting at $30,000. Prices are figured depending on Epoxy fill, end weight, and size. 
    [tags] => Array
        (
            [0] => art
        )

    [price] => 20000
    [shipping_profile_id] => 64924870035
    [shop_section_id] => 26873940
    [is_taxable] => 
    [taxonomy_id] => 12213
    [who_made] => i_did
    [when_made] => made_to_order
    [is_supply] => 
    [is_personalizable] => 1
    [personalization_is_required] => 
    [personalization_char_count_max] => 256
    [personalization_instructions] => Please enter any custom details
    [is_customizable] => 1
    [should_auto_renew] => 
    [processing_min] => 1
    [processing_max] => 3
)
```

**Step to reproduce:**
- Create a new listing with a calculated shipping profile using this [createDraftListing](https://developers.etsy.com/documentation/reference/#operation/createDraftListing) API endpoint.

**Additional_details:**
```
[shop_id]=> 17815748
[X-Etsy-Request-Uuid] => Array
                        (
                            [0] => EutTv55__DlrbHOH_3-a1_DzGo4a
                        )
```
@etsyachristensen If any other data is required. please let me Know
Thanks!

**Comment by etsyachristensen** [comment]
@Deepshikha-cedcoss We are moving this issue to our Discussion board since we are closing down the issue board.
